source/draw/RasterMonochrome.ooc

```diff
@@ -41,14 +41,12 @@ RasterMonochrome: class extends RasterPacked {
 		this apply(ColorConvert fromMonochrome(action))
 	}
 	apply: func ~monochrome (action: Func(ColorMonochrome)) {
-		end := this buffer pointer as Long + this buffer size
-		rowLength := this size width * this bytesPerPixel
-		for (row in this buffer pointer as Long .. end) {
-			rowEnd := row + rowLength
-			for (source: Long in row .. rowEnd)
-				action((source as ColorMonochrome*)@)
-			row += this stride - 1
-		}
+		pointer := this buffer pointer as ColorMonochrome*
+		for (row in 0 .. this height)
+			for (col in 0 .. this width) {
```

Comment:
I might merge this before you have a chance to fix it, but do try to keep in mind to type out the full name of things, at least unless the result is ridiculously long. `col` should be `column`.